### PR TITLE
tools: remove all deprecated options from the help output

### PR DESCRIPTION
**Pull request by zhengchengyao**
Signed-off-by: Zheng Yin <zhengyin@cmss.chinamobile.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by zhengchengyao** [comment]
Hi @dillaman, Please help take a review. I will remove all deprecated options from the help output in next PR,  thanks.

**Comment by dillaman** [comment]
@zhengchengyao Can you just do it all in one PR since the goal is to remove those help statements, so merging this PR just to have a future one remove the changes from this PR seems counterproductive.

**Comment by zhengchengyao** [comment]
OK.

**Comment by zhengchengyao** [comment]
Hi @dillaman 
I'm not sure whether I need to delete deprecated options in  src/test/cli/rbd/help.t. 
thanks

File: src/tools/rbd/OptionPrinter.cc [hunk 1]
```diff
@@ -74,9 +74,23 @@ void OptionPrinter::print_detailed(std::ostream &os) {
     std::cout << std::endl;
   }
 
+  std::vector<std::string> deprecated_set = {"order", "new-format"};
```
Comment (dillaman) [inline src/tools/rbd/OptionPrinter.cc]: I would prefer that you don't add a list of deprecated options. Instead, if the help text includes `deprecated`, don't print the line. Similarly, upon parsing the command line, if someone specifies a command w/ "deprecated" in the help test, print the warning.

File: src/tools/rbd/Shell.cc [hunk 2]
```diff
@@ -343,9 +343,29 @@ void Shell::print_help() {
     }
   }
 
-  po::options_description global_opts(OptionPrinter::OPTIONAL_ARGUMENTS);
+  po::options_description global_opts;
   get_global_options(&global_opts);
-  std::cout << std::endl << global_opts << std::endl
+  std::string indent2(2, ' ');
+
+  std::cout << std::endl << OptionPrinter::OPTIONAL_ARGUMENTS << ":" << std::endl;
```
Comment (dillaman) [inline src/tools/rbd/Shell.cc]: Why can't this be part of `OptionPrinter`?

**Comment by zhengchengyao** [comment]
Hi @dillaman, I use deprecated option, I will output warning.

for example:

root@dev:/data/ceph/build# ./bin/rbd create volumes/test -s 1G --order 22 --new-format

rbd: --new-format is deprecated, use --image-format
rbd: --order is deprecated, use --object-size

Do I need add "warning" in output ?  Maybe like this.
warning: rbd: --new-format is deprecated, use --image-format

thanks.

**Comment by dillaman** [comment]
> Hi @dillaman, I use deprecated option, it will output warning.
> 
> for example:
> 
> root@dev:/data/ceph/build# ./bin/rbd create volumes/test -s 1G --order 22 --new-format
> 
> rbd: --new-format is deprecated, use --image-format
> rbd: --order is deprecated, use --object-size

That is the type of warning I am talking about -- but instead of hard-coding them throughout the code, I am asking for a centralized piece of code that can output these generically. Perhaps if you change the help text for deprecated messages to `deprecated[:<new parameter name>]`, if you find a string after the colon, you can output `, use --<new parameter name>` in addition to `rbd: --<parameter> is deprecated`.

File: src/test/cli/rbd/help.t [hunk 3]
```diff
@@ -151,15 +151,14 @@
...
-    --user arg            client id (without 'client.' prefix)
-    -n [ --name ] arg     client name
-    -m [ --mon_host ] arg monitor host
-    --secret arg          path to secret key (deprecated)
-    -K [ --keyfile ] arg  path to secret key
-    -k [ --keyring ] arg  path to keyring
+    -c [ --conf ] arg                   path to cluster configuration
+    --cluster arg                       cluster name
+    --id arg                            client id (without 'client.' prefix)
+    --user arg                          client id (without 'client.' prefix)
```
Comment (dillaman) [inline src/test/cli/rbd/help.t:157]: Nit: I think `--user` is essentially deprecated as well.

File: src/tools/rbd/Shell.cc [hunk 4]
```diff
@@ -202,6 +202,25 @@ int Shell::execute(int argc, const char **argv) {
     }
 
     int r = (*action->execute)(vm, ceph_global_init_args);
+
+    if (vm.size() > 0) {
+      for (auto opt : vm) {
+	try {
+          auto option = command_opts.find(opt.first, false);
```
Comment (dillaman) [inline src/tools/rbd/Shell.cc:209]: Nit: indentation is off

Comment (zhengchengyao) [reply src/tools/rbd/Shell.cc:209]: done

File: src/test/cli/rbd/help.t (hunk 3 above)
Comment (zhengchengyao) [reply src/test/cli/rbd/help.t]: done

**Comment by zhengchengyao** [comment]
Hi @dillaman , Updated, Please help review it, thanks

**Pull request merged**